staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -105,6 +108,47 @@ func (mounter *Mounter) hasSystemd() bool {
 	return *mounter.withSystemd
 }
 
+// Do a bind mount including the needed remount for applying the bind opts.
+// If the remount fails and the KubeletInUserNamespace feature gate is enabled
+// figure out if the source filesystem has the ro, nodev, noexec, nosuid,
+// noatime, relatime or nodiratime flag set and try another remount with the found flags.
+func (mounter *Mounter) bindMountSensitive(mounterPath string, mountCmd string, source string, target string, fstype string, bindOpts []string, bindRemountOpts []string, bindRemountOptsSensitive []string, mountFlags []string, systemdMountRequired bool) error {
+	err := mounter.doMount(mounterPath, defaultMountCommand, source, target, fstype, bindOpts, bindRemountOptsSensitive, mountFlags, systemdMountRequired)
+	if err != nil {
+		return err
+	}
+	err = mounter.doMount(mounterPath, defaultMountCommand, source, target, fstype, bindRemountOpts, bindRemountOptsSensitive, mountFlags, systemdMountRequired)
+	if utilfeature.DefaultFeatureGate.Enabled("KubeletInUserNamespace") {
```

Comment:
Probably you should better check whether the kubelet is actually running in UserNamespace, not just whether the feature gate is enabled:
- https://github.com/search?q=repo%3Akubernetes%2Fkubernetes+RunningInUserNS&type=code

The feature gate can be just removed.

Reply:
Thanks for the hint. Just implemented.